### PR TITLE
[Nix] use specific version of nixpkgs

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by hydai** [comment]
Should we have a CI to verify this?

**Comment by dannypsnl** [comment]
> Should we have a CI to verify this?

Yes, I think so, we were using unstable before and cause random failure, but if I fix a version then it should be more stable and controllable.

**Comment by dannypsnl** [comment]
Also, a linker issue will show if set BUILD_TYPE to Release, below is the error message
```sh
ld: warning: ignoring file CMakeFiles/wasmedgeCAPI.dir/wasmedge.cpp.o, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0xDE 0xC0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0x18 0xB4 0x0A 0x00 )
ld: warning: ignoring file ../driver/libwasmedgeDriver.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../plugin/libwasmedgePlugin.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../plugin/wasi_logging/libwasmedgePluginWasiLogging.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../vm/libwasmedgeVM.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../validator/libwasmedgeValidator.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../host/wasi/libwasmedgeHostModuleWasi.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../loader/libwasmedgeLoaderFileMgr.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../po/libwasmedgePO.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../executor/libwasmedgeExecutor.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../loader/libwasmedgeLoader.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../../thirdparty/blake3/libutilBlake3.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../system/libwasmedgeSystem.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../llvm/libwasmedgeLLVM.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../common/libwasmedgeCommon.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
ld: warning: ignoring file ../aot/libwasmedgeAOT.a, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x23 0x31 0x2F 0x31 0x32 0x20 0x20 0x20 )
```

We can find the file types
```sh
$ echo -e '\xde\xc0\x17\x0b' | file -
/dev/stdin: LLVM bitcode, wrapper
$ echo -e '\x21\x3c\x61\x72\x63\x68\x3e\x0a\x23\x31\x2f\x31' | file -
/dev/stdin: current ar archive
```

**Review by 0yi0 — CHANGES_REQUESTED** [review]
I see `"-DWASMEDGE_BUILD_PLUGINS=OFF"`, so please update paths as #3678 did, thanks.

File: .github/workflows/build_for_nix.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  pull_request:
+    branches:
+      - master
+      - "proposal/**"
+    paths:
+      - "flake.nix"
+      - ".github/workflows/build_for_nix.yml"
+      - "include/**"
+      - "lib/**"
+      - "plugins/**"
```
Comment (0yi0) [inline .github/workflows/build_for_nix.yml:27]: also here

File: .github/workflows/build_for_nix.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - "test/**"
```
Comment (0yi0) [inline .github/workflows/build_for_nix.yml:28]: and here

**Review by 0yi0 — CHANGES_REQUESTED** [review]

**Review by 0yi0 — DISMISSED** [review]

File: .github/workflows/build_for_nix.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  push:
+    branches:
+      - master
+    paths:
+      - "flake.nix"
+      - "flake.lock"
+      - ".github/workflows/build_for_nix.yml"
+      - "include/**"
+      - "lib/**"
+      - "!test/**"
```
Comment (0yi0) [inline .github/workflows/build_for_nix.yml:13]: Oh wait, if just remove the 4 lines with `test/**` if you don't want to watch on tests at all lol

File: flake.nix [hunk 4]
```diff
@@ -1,83 +1,49 @@
...
       in with pkgs; rec {
         packages = { wasmedge = wasmedge; };
         packages.default = packages.wasmedge;
         devShells.default = mkShell {
           buildInputs = [
             wasmedge
-
             ninja
-            rust
             gcovr
```
Comment (hydai) [inline flake.nix:45]: Do we run the coverage tests with nix? If not, let's remove this dependency.

**Review by hydai — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 4 above)
Comment (dannypsnl) [reply flake.nix:45]: My understanding is this is for development, so user can still build test in it again.

Comment (hydai) [reply flake.nix:45]: I see.

**Review by hydai — APPROVED** [review]
LGTM. @0yi0 PTAL

**Comment by hydai** [comment]
And we should squash this PR to make it a clear commit for this feature.

**Review by 0yi0 — APPROVED** [review]

**Pull request merged**